peer/peer.go

```diff
@@ -1967,7 +1964,7 @@ func (p *Peer) start() error {
 // have no effect.
 func (p *Peer) AssociateConnection(conn net.Conn) {
 	// Already connected?
-	if !atomic.CompareAndSwapInt32(&p.connected, 0, 1) {
+	if !p.connected.CompareAndSwap(false, true) {
```

Comment:
I like how `CompareAndSwap` was being used here to ensure actions are only performed once. I have copied this pattern to other places where it is appropriate (once in this file, twice in `addrmanager.go` and once in `rpcwebsocket.go`)